app/views/bookmarks/_filters.html.erb

```diff
@@ -128,6 +128,7 @@
         </dl>
       </dd>
       <dt class="landmark"><%= ts("Submit") %></dt>
+      <dd class="submit actions"><%= f.submit ts("Clear Filters") %></dd>
```

Comment:
Instead of a callback that redirects, I think this button can be a link to whatever the original form / starting URL was.

For the original form of the works and bookmarks filters, you'll have to handle 5 different kinds of owning objects:

https://github.com/otwcode/otwarchive/blob/808b34b1d21be65c26fc9ae4a20621c40955a37d/app/views/bookmarks/_filters.html.erb#L135-L139

You can add a helper to determine the original URL in app/helpers/search_helper.rb.

----

The original form of the collections filters can just be `collections_path`. On that note, can you also change this part to `collections_path`?

https://github.com/otwcode/otwarchive/blob/808b34b1d21be65c26fc9ae4a20621c40955a37d/app/views/collections/_filters.html.erb#L1

----

There's one more filters sidebar at app/views/works/_collection_filters.html.erb. The reset path should just be `collected_user_works_path(@user)`.

Reply:
Thank you for explaining!